modules/core/src/main/scala/org/tessellation/infrastructure/dag/DAGService.scala

```diff
@@ -55,5 +55,20 @@ object DAGService {
 
       }
 
+      def getTotalWallets: F[Option[(Int, SnapshotOrdinal)]] =
+        globalSnapshotStorage.head.flatMap {
+          _.flatTraverse(snapshot => getTotalWallets(snapshot.value.ordinal))
+        }
+
+      def getTotalWallets(ordinal: SnapshotOrdinal): F[Option[(Int, SnapshotOrdinal)]] =
```

Comment:
This function receives ordinal and then returns it, which is odd. I think you don't need it at all. Once you have `head` you can do just `head.info.balances.size`.

Reply:
Ok, I get it. Let's keep it then.